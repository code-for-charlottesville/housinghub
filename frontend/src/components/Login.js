import React from "react";

import "../style/App.css";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import { setLoginField, loginUser } from "../actions/login";
import { LoadingSpinner } from "./LoadingSpinner";
import { Container, Row, Col, Form, Button } from "react-bootstrap";

const Login = (props) => {
  if (props.isLoggedIn) return <Redirect to="/home" />;
  return (
    <Container>
      <Row className="login">
        <Col>
          <h1>Housing Hub</h1>
          {props.loading && <LoadingSpinner />}
          {!props.loading && (
            <Form>
              <Form.Group controlId="formBasicEmail">
                <Form.Control
                  type="email"
                  placeholder="Email"
                  value={props.fields.email}
                  onChange={(e) =>
                    props.dispatch(setLoginField("email", e.target.value))
                  }
                />
              </Form.Group>

              <Form.Group controlId="formBasicPassword">
                <Form.Control
                  type="password"
                  placeholder="Password"
                  value={props.fields.password}
                  className="input"
                  onChange={(e) =>
                    props.dispatch(setLoginField("password", e.target.value))
                  }
                />
              </Form.Group>
              <Form.Group controlId="formBasicCheckbox">
                <Form.Check
                  type="checkbox"
                  label="Remember Me"
                  checked={props.fields.rememberMe}
                  onChange={() =>
                    props.dispatch(
                      setLoginField("rememberMe", !props.fields.rememberMe)
                    )
                  }
                />
              </Form.Group>
              <Button
                variant="primary"
                type="submit"
                onClick={() => loginUser()}
              >
                Login
              </Button>
            </Form>
          )}
        </Col>
      </Row>
    </Container>
  );
};

function mapStateToProps(state) {
  return {
    fields: state.login.fields,
    isLoggedIn: state.login.isLoggedIn,
    loading: state.appState.loading,
    error: state.login.error,
  };
}

export default connect(mapStateToProps)(Login);
