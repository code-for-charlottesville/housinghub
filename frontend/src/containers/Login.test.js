import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import sampleMockStore from "../../test/mockStore.js";
import Login from "./Login";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

const mockStore = configureStore([]);

describe("containers/Login", () => {
  let _matchesSnapshot = (C, props = {}) => {
    let store = mockStore({
      ...sampleMockStore,
      appState: {
        ...sampleMockStore.appState,
        loading: props.loading,
      },
      login: {
        ...sampleMockStore.login,
        ...props,
      },
    });
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={["/my/initial/route"]}>
          <Provider store={store}>
            <C {...props} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  };

  let testTable = [
    {
      name: "renders fields",
      component: Login,
      props: {
        fields: {
          rememberMe: true,
          password: "test",
          email: "user@gmail.com",
        },
        isLoggedIn: false,
        loading: false,
        error: "sample error",
      },
    },
    {
      name: "renders nothing is already logged in",
      component: Login,
      props: {
        isLoggedIn: true,
      },
    },
    {
      name: "loading spinner if loading",
      component: Login,
      props: {
        isLoggedIn: false,
        loading: true,
      },
    },
  ];

  testTable.forEach((t) => {
    it(t.name, () => {
      _matchesSnapshot(t.component, t.props);
    });
  });
});
