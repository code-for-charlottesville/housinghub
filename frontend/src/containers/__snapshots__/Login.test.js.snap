// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Login loading spinner if loading 1`] = `
<div
  className="container"
>
  <div
    className="login row"
  >
    <div
      className="col"
    >
      <h1>
        Housing Hub
      </h1>
      <div
        className="spinner-border"
        role="status"
      />
    </div>
  </div>
</div>
`;

exports[`containers/Login renders fields 1`] = `
<div
  className="container"
>
  <div
    className="login row"
  >
    <div
      className="col"
    >
      <h1>
        Housing Hub
      </h1>
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="formBasicEmail"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value="user@gmail.com"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="input form-control"
            id="formBasicPassword"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value="test"
          />
        </div>
        <div
          className="form-group"
        >
          <div
            className="form-check"
          >
            <input
              checked={true}
              className="form-check-input"
              disabled={false}
              id="formBasicCheckbox"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="form-check-label"
              htmlFor="formBasicCheckbox"
              title=""
            >
              Remember Me
            </label>
          </div>
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`containers/Login renders nothing is already logged in 1`] = `null`;
